src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,324 @@
+import { fork } from "child_process";
+import createInstanceName from "./process-name";
+import envPaths from "env-paths";
+import psList, { ProcessDescriptor } from "@trufflesuite/ps-list";
+import { Dirent, promises as fsPromises } from "fs";
+// this awkward import is required to support node 12
+const { readFile, rm, mkdir, readdir, writeFile, unlink } = fsPromises;
+import path from "path";
+import { FlavorName } from "@ganache/flavors";
+
+export type DetachedInstance = {
+  name: string;
+  pid: number;
+  startTime: number;
+  host: string;
+  port: number;
+  flavor: FlavorName;
+  cmd: string;
+  version: string;
+};
+
+const FILE_ENCODING = "utf8";
+const READY_MESSAGE = "ready";
+const START_ERROR =
+  "An error occurred spawning a detached instance of Ganache:";
+const dataPath = envPaths(`Ganache/instances`).data;
```

Comment:
We don't uses a suffix for the persistent cache `db`, let's also drop it here:

```suggestion
const dataPath = envPaths("Ganache/instances", {
  suffix: ""
}).data;
```

Reply:
My instances!